File: Core Java/doogdoor-app/src/com/techlabs/dogdoor/BarkRecognizer.java
package com.techlabs.dogdoor;

import java.util.List;

public class BarkRecognizer {
	private DogDoor door;

	public BarkRecognizer(DogDoor door) {
		this.door = door;
	}

	public void recognize(Bark bark) {
		System.out.println("   BarkRecognizer: Heard a '" + bark.getSound()
				+ "'");
		List<Bark> allowedBarks = door.getAllowedBarks();
		for (Bark b : allowedBarks) {
			Bark allowedBark = b;
			if (allowedBark.equals(bark)) {
				door.open();
				return;
			}
		}
		System.out.println("This dog is not allowed.");
	}

}

File: DOTNET/C#/DogDoorApp/DogDoorApp/BarkRecognizer.cs
using System;
using System.Collections.Generic;

namespace DogDoorApp
{
    public class BarkRecognizer
    {
        private DogDoor door;

        public BarkRecognizer(DogDoor door)
        {
            this.door = door;
        }

        public void Recognize(Bark bark)
        {
            Console.WriteLine("   BarkRecognizer: Heard a '" + bark.Sound
                + "'");
            List<Bark> allowedBarks = door.AllowedBark;
            foreach (Bark b in allowedBarks)
            {
                Bark allowedBark = b;
                if (allowedBark.Equals(bark))
                {
                    door.Open();
                    return;
                }
            }
            Console.WriteLine("This Dog is not Allowed..");
        }
    }
}
